### PR TITLE
Create library-build.yml

### PR DESCRIPTION
**Pull request by AmalieDue**
An initial workflow file for the library has been added.

### Type(s) of Change (Split the PR if you check many)
- [ ] Added user feature
- [ ] Added technical feature
- [ ] Added tests
- [ ] Refactor
- [ ] Bugfix
- [ ] Updated documentation
- [ ] Kubernetes changes
- [x] CI/CD workflow changes

### Testing Checklist
- [ ] I have added/updated unit tests for functional changes.
- [ ] I have added/updated integration tests for changes that affect dependent systems.
- [ ] I have added/updated end-to-end tests for feature changes.

### General Checklist
- [ ] I have pulled in the latest changes from master.
- [ ] I have run `make lint`.
- [ ] I have successfully run `make tests`.
- [x] I have updated relevant CI/CD workflows.
- [ ] I have updated relevant Kubernetes manifests/scripts.
- [ ] I have updated/added relevant (internal and external) documentation (readme, doc comments, etc).
- [ ] I have updated the dependency map to reflect my changes.
- [ ] I have added the license to new source code files.
- [ ] I have spent some time looking over the full diff before creating this PR.

### CODE REVIEW

**Comment by AmalieDue** [comment]
I have created a library workflow that builds, lints, and run tests.

File: lib/makefile [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# limitations under the License.
+
+##### Help message #####
+help:  ## Display this help
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make <target> \033[36m\033[0m\n\nTargets:\n"} /^[a-zA-Z0-9_-]+:.*?##/ { printf "  \033[36m%-20s\033[0m %s\n", $$1, $$2 }' $(MAKEFILE_LIST)
+
+##### Build targets #####
+.PHONY: build
+build: ## Build the Encryptonize library
+	go get -v ./...
```
Comment (philip-cc) [inline lib/makefile:22]: This shouldn't be necessary.
```suggestion
```

File: lib/makefile [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	go build -v .
```
Comment (philip-cc) [inline lib/makefile:23]: No need for `-v` here I'd say. 
```suggestion
	go build .
```

File: lib/makefile [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+.PHONY: lint
+lint: ## Lint the codebase
+	echo "[*] formatting code"
+	gofmt -l -w .
+
+	echo "[*] tidying up"
+	go mod tidy
+
+	echo "[*] running linter"
```
Comment (philip-cc) [inline lib/makefile:33]: Make prints out what it does, so the echos are superfluous. 
```suggestion
	gofmt -l -w .
	go mod tidy
```

File: lib/makefile [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+	golangci-lint run
```
Comment (philip-cc) [inline lib/makefile:34]: We typically enable a bit more linters here:
```suggestion
	golangci-lint run -E gosec,asciicheck,bodyclose,gocyclo,unconvert,gocognit,misspell,revive,whitespace --timeout 5m
```

File: lib/makefile [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+##### Test targets #####
+.PHONY: tests
+tests: ## Run tests
+	go test -count=1
```
Comment (philip-cc) [inline lib/makefile:39]: This doesn't run tests on the nested packages.
```suggestion
	go test -count=1 ./...
```

File: .github/workflows/library-build.yml [hunk 6]
```diff
@@ -0,0 +1,53 @@
+name: Library build
+
+on:
+  push:
+    branches: AmalieDue-patch-1
```
Comment (philip-cc) [inline .github/workflows/library-build.yml:5]: Remember to remove this before merging.

File: .github/workflows/library-build.yml [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+    - name: Setup and cache dependencies
+      uses: cyber-crypt-com/encryptonize-core/.github/actions/encryption-service-build-setup@master
+
+  lib-build:
+    needs: setup
+    runs-on: ubuntu-20.04
+    defaults:
+      run:
+        working-directory: lib
```
Comment (philip-cc) [inline .github/workflows/library-build.yml:23]: You could put this as a top level default.

Comment (AmalieDue) [reply .github/workflows/library-build.yml]: fixed

**Review by philip-cc — DISMISSED** [review]
Ship it!

**Comment by AmalieDue** [comment]
It seems like I need a new approval

**Review by philip-cc — APPROVED** [review]
Looks good

**Pull request merged**